Add a contributing section to the documentation (#497)
containous/maesh: docs/content/contributing/building-testing.md

# Building and Testing

So you want to build your own Maesh binary from the sources? Let's see how.

## Building

To build Maesh from the sources you need either [Docker](https://github.com/docker/docker) and [make](https://www.gnu.org/software/make/manual/make.html), 
or [Go](https://github.com/golang/go). 

### With `Docker` and `make`

Maesh can be built from the sources by using the `make` command. This will create a binary for the Linux platform in 
the `dist` directory and a Docker image:

```bash
$ make
#[...]
Successfully tagged containous/maesh:latest
docker run --name=build -t "containous/maesh:latest" version
version:
 version     : 04789a3
 commit      : 04789a3
 build date  : 2020-04-09_10:24:49AM
 go version  : go1.14.1
 go compiler : gc
 platform    : linux/amd64
#[...]

$ ls dist/
maesh
``` 

!!! Note
    The default `make` target invokes the `clean`, `check`, `test` and `build` targets.

### With `Go`

Requirements:

- `Go` v1.14+

> - Environment variable `GO111MODULE=on`

Reviewer: Really? Go1.14+ it shouldn't make a difference, no?

Author: After a quick test the project cannot be built with a version earlier than Go `1.14` because of Traefik dependency. Here is the build output:

```
github.com/containous/traefik/v2/pkg/tls
../../../.gvm/pkgsets/go1.13/global/pkg/mod/github.com/containous/traefik/v2@v2.2.0/pkg/tls/cipher.go:32:52: undefined: tls.TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256
../../../.gvm/pkgsets/go1.13/global/pkg/mod/github.com/containous/traefik/v2@v2.2.0/pkg/tls/cipher.go:34:52: undefined: tls.TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256
../../../.gvm/pkgsets/go1.13/global/pkg/mod/github.com/containous/traefik/v2@v2.2.0/pkg/tls/cipher.go:64:3: undefined: tls.TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256
../../../.gvm/pkgsets/go1.13/global/pkg/mod/github.com/containous/traefik/v2@v2.2.0/pkg/tls/cipher.go:65:3: undefined: tls.TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256
note: module requires Go 1.14
```

Reviewer: Ah, okay! Then you're fine.

> One you've setup your Go environment you can build Maesh from the sources by using the `go build` command. The Go 

Reviewer 2: ```suggestion
Once your Go environment is set up, you can build Maesh from the sources by using the `go build` command. The Go 
```

> compiler will build a binary for your platform which can be run via a shell.

Reviewer 2: ```suggestion
compiler will build an executable binary for your platform.
```
